unittests/gettestkernels.py

```diff
@@ -22,38 +22,62 @@
 SOFTWARE.
 """
 
+import hashlib
 import os
-import time
-import platform
-import tempfile
-import urllib
-import urllib.error
-import urllib.request
+import shutil
 import sys
-import hashlib
+import tempfile
+import time
+from pathlib import Path
+
+import requests
+from requests import RequestException
 
-try:
-    cwd = os.environ['CSPYCE_TEST_KERNELS']
-except KeyError:
-    cwd = "/tmp" if platform.system() == "Darwin" else tempfile.gettempdir()
 
+def _get_kernel_directory():
+    try:
+        return os.environ['CSPYCE_TEST_KERNELS']
+    except KeyError:
+        pass
+
+    directory = Path(tempfile.gettempdir()) / "cspyce-test-kernels"
+    directory.mkdir(parents=True, exist_ok=True)
+    return str(directory)
+
+
+def _get_test_output_directory():
+    directory = Path(tempfile.gettempdir()) / "test_file_output"
```

Comment:
Let's be consistent about dashes (cspyce-test-kernels) vs underscores (test_file_output). Dashes seem to be more common in filenames. Also let's put "cspyce" in the test-file-output name just so it doesn't conflict with any other program.